webpack/assets/javascripts/react_app/routes/ExternalRoutes.js

```diff
@@ -0,0 +1,23 @@
+import { Route } from 'react-router-dom';
+import React from 'react';
+import { addGlobalFill } from '../components/common/Fill/GlobalFill';
+import { updateLocation } from '../routes';
+
+export const addGlobalRoutes = (id, routes) =>
+  routes.map(({ render, path, ...routeProps }, index) =>
+    addGlobalFill(
+      'routes',
+      `${id}-${index}`,
+      <Route
+        path={path}
+        key={path}
+        {...routeProps}
+        render={renderProps => {
+          const railsContainer = document.getElementById('rails-app-content');
+          if (railsContainer) railsContainer.remove();
+          renderProps.location && updateLocation(renderProps.location);
+          return render(renderProps);
```

Comment:
Would it make sense to create a new file call `RoutingService.js` that will handle those type of logics?
It can also contain and manage the `currentLocation` in the `routes/index.js`

Reply:
totally make sense :+1: